xdsl/irdl/constraints.py

```diff
@@ -179,14 +129,18 @@ def verifies(self, attr: Attribute) -> TypeGuard[AttributeCovT]:
         except VerifyException:
             return False
 
-    def get_variable_extractors(self) -> dict[str, VarExtractor[Attribute]]:
+    def variables(self) -> dict[str, ConstraintVarType]:
         """
-        Get a dictionary of constraint variables to extractors for these variables,
-        which provide a method to obtain the value of each constraint variable from
-        the value of the attribute which is verified by this constraint.
+        Returns a dictionary of the variables that can be extracted by this constraint.
         """
         return {}
 
+    def extract_var(self, attr: Attribute, var: str) -> ConstraintVariableType:
+        """
+        Extracts the value of an constraint variable `var` from the input attribute `attr`.
+        """
+        raise ValueError("Cannot extract variable from constraint")
```

Comment:
It would be nice to give as much info for debugging as possible in these errors, otherwise one has to add them manually when it's raised.